### PR TITLE
Return the foreign function result.

### PR DESCRIPTION
**Pull request by jplevyak**
Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

**Comment by mandarjog** [comment]
needs to be cherry-picked to 1.5

**Comment by jplevyak** [comment]
Responding to Piotr.   The foreign function API has provision for returning an error.  If there is an error which fits in the WasmResult semantics, I think it is reasonable to include it instead of the OK.   I agree that it does change the semantics, but it is not unreasonable.   Let us discuss this on the ABI document and I will merge this and if we have to change it as part of upstreaming or the ABI document review we can do it then.

File: source/extensions/common/wasm/exports.cc [hunk 1]
```diff
@@ -248,7 +248,7 @@ Word call_foreign_function(void* raw_context, Word function_name, Word function_
   if (!context->wasmVm()->setWord(results_size.u64_, Word(result_size))) {
     return wasmResultToWord(WasmResult::InvalidMemoryAccess);
   }
```
Comment (PiotrSikora) [inline source/extensions/common/wasm/exports.cc]: If `res != WasmResult::Ok` then you probably shouldn't be writing the `result` above.

Comment (PiotrSikora) [reply source/extensions/common/wasm/exports.cc]: @jplevyak could you fix that? Since this is already merged.

Comment (jplevyak) [reply source/extensions/common/wasm/exports.cc]: Perhaps.  Again.  I will defer to reviewers.  I am OK providing more information to the caller assuming that the foreign function provided it as this that is part of a private API between the caller and the callee.

Comment (kyessenov) [reply source/extensions/common/wasm/exports.cc]: It's fine to pass error details.

Comment (PiotrSikora) [reply source/extensions/common/wasm/exports.cc]: Usually, when call fails (i.e. `res != WasmResult::Ok`), it means that results were not necessarily initialized, and `result` and `result_size` may point to garbage.

**Pull request merged**